guides for inference in RL/Verl (#2055)
llm-d/llm-d: guides/rl/README.md

is the problem llm-d and its Endpoint Picker (EPP) were built to solve for
production serving, and the same machinery applies directly to RL rollout.

The following guides describe two ways to improve inference & rollout in an RL
training loop.

## [Leverage llm-d and EPP directly in your RL framework](./llm-d-verl-integration.md)

llm-d's [no-Kubernetes mode](../no-kubernetes-deployment/README.md) uses
the routing stack - EPP, Envoy, and the vLLM workers - as plain processes
rather than Kubernetes resources, so it drops cleanly into a Ray or Slurm
cluster without requiring a control plane. This lets an RL framework replace
its naive rollout routing with EPP through configuration alone, with no
changes to the framework's source. Using llm-d's EPP brings the benefits of
advanced routing - KV-cache-aware routing, session stickiness, burst-aware
routing, and more. Likewise, using llm-d as the serving backend brings the
benefits of the full inference stack - P/D (prefill/decode) disaggregation,
KV-cache offloading, and more - a meaningful throughput win for the large,
prefix-sharing sample groups that GRPO and PPO produce.

Two integration modes are supported:

> - **EPP as the endpoint picker** - the framework calls EPP directly to pick a
>   replica, then dispatches the rollout request itself.

Reviewer: ```suggestion
- **llm-d EPP** - the framework calls the EPP directly to pick a specific model server, then dispatches the rollout request itself.
```

- **llm-d as an inference backend endpoint** - the framework sends generation

>   to a single llm-d serving endpoint that consults EPP and forwards it.

Reviewer: We have a canonical name for "inference backend endpoint", `llm-d Router`

> - **llm-d as an inference backend endpoint** - the framework sends generation
>   to a single llm-d serving endpoint that consults EPP and forwards it.

Reviewer: ```suggestion
- **llm-d Router** - the framework sends generation
  to an llm-d Router that consults EPP for request routing.
```
